llm/configuration.go

```diff
@@ -71,8 +69,14 @@ type BotConfig struct {
 }
 
 func (c *BotConfig) IsValid() bool {
-	// Basic validation
-	if c.Name == "" || c.DisplayName == "" || c.Service.Type == "" {
+	// Basic validation - service validation happens separately
+	// Note: ServiceID can be empty if Service is embedded (deprecated)
```

Comment:
This should be removed in the follow up PR, it's not.

Reply:
Done here: https://github.com/mattermost/mattermost-plugin-agents/pull/411/commits/876003813608274984ccf4987cea5e760d64e5a1